python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -540,6 +541,10 @@ def ListRepositories(
                     container_image=self._container_image,
                     container_context=self._container_context,
                     dagster_library_versions=DagsterLibraryRegistry.get(),
+                    can_create_snapshots_in_run_worker={
+                        repo_name: repo_def.repository_load_data is None
+                        for repo_name, repo_def in loaded_repositories.definitions_by_name.items()
+                    },
```

Comment:
@alangenfeld one potential short-term workaround for the repository_load_data issue here (i originally added this field for back-compat with old code, now it excludes any repos with repository_load_data)

Longer term we would probably want to give repository_load_data its own snapshot type and decouple it from execution plan - since unlike execution plan, we can store it in the host cloud and attach it to the run